### DISCUSSION TITLE
Umi4 的 .umirc.ts  defineConfig 在 VSCode 中  没有智能提示.

### DISCUSSION THREAD

**Discussion by xxxxue** [umijs/umi#9167] [General]
[官网文档](https://umijs.org/docs/guides/typescript)
`.umirc.ts`  的 `defineConfig`  在 VSCode 中 没有智能提示.  (在 Jetbrains IDE 中是有的)

手动导出两个类型, 有智能提示了

具体情况如下图:

![image](https://user-images.githubusercontent.com/32764266/187423799-7a7429cd-281c-46df-8afc-e2ab8fed6500.png)

- **Umi Version**: v4.0.15
- **Node Version**: v16.14.2
- **Platform**: Windows 11
- **VSCode**: v1.70.0

补充说明:
pnpm 需要在  `.npmrc` 中配置 `public-hoist-pattern[]="@umijs/preset-umi"`

**Comment by fz6m** [comment]
确实有问题，但不应该用提升包的方式解。
